designcompose/src/main/java/com/android/designcompose/FrameRender.kt

```diff
@@ -154,6 +156,7 @@ private fun calculateProgressBarData(
     progressBarData: ProgressBarMeterData,
     meterValue: Float,
     style: ViewStyle,
+    parent: SquooshResolvedNode?,
```

Comment:
I was thinking it's a bit cleaner to just pass `parentLayout: Layout?` in both of these, but you're consulting style to see if the layout is fixed, so you wouldn't be able to do that then.

Do you need to check if it's fixed, or could you always use `parentLayout` if it's not null (understanding that it is null on DC Classic)?

Reply:
That's exactly what I did originally, but it breaks in squoosh on progress bars that are skewed or rotated with a fixed size (no constraints to stretch). In that scenario, we want to use the non-rotated width/height, which is saved in the plugin data. Using the width/height from `computedLayout` yields a smaller value and renders incorrectly. However if the progress bar is modified in Figma to stretch (constraints top/bottom or left/right), then `computedLayout` has the correct size that we want to use. I'll need to dig a bit to figure out why exactly layout is computed this way, I don't recall exactly and am not sure if it is different in squoosh.

Reply:
This is because of the complicated nature of how we determine the size of a node when rendering, with most of the logic in `getNodeRenderSize()`. To make this more robust I'll try to re-use this function instead of defining additional logic here.

Reply:
Done